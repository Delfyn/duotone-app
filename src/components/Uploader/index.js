/* @flow */

import React, { Component } from 'react';
import { createDuotoneImage } from 'react-duotone';
import UploaderInput from '../UploaderInput';
import UploaderColors from '../UploaderColors';
import Button from '../Button';
import './style.css';

type Props = {
  originalFile: any,
  primary: string,
  secondary: string,
};

class Uploader extends Component {
  constructor(props: Props) {
    super(props);
    this.state = {
      downloadDisabled: true,
    };
    this.onDownloadClick = this.onDownloadClick.bind(this);
  }
  state = {
    downloadDisabled: Boolean,
  };
  onDownloadClick: () => void;
  onDownloadClick() {
    const { originalFile, primary, secondary } = this.props;
    const img = new Image();
    img.src = originalFile;
    img.onload = () => {
      const duotoneImage = createDuotoneImage(img, primary, secondary);
      const downloadLink = document.createElement('a');
      downloadLink.href = duotoneImage;
      downloadLink.download = 'duotone-export.png';
      downloadLink.target = '_blank';
      downloadLink.style.display = 'none';
      if (document.body) {
        document.body.appendChild(downloadLink);
        downloadLink.click();
      }
      if (document.body && document.body.removeChild) {
        document.body.removeChild(downloadLink);
      }
    };
  }
  render() {
    return (
      <div>
        <UploaderInput {...this.props} />
        <UploaderColors {...this.props} />
        <footer className="flex px2 pb2">
          <Button
            className="flex-auto"
            disabled={this.props.downloadDisabled}
            onClick={this.onDownloadClick}>
            Download
          </Button>
        </footer>
      </div>
    );
  }
}

export default Uploader;
