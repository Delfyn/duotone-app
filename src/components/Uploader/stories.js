/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import Uploader from './index';
import Panel from '../Panel';

const UploaderWithProps = ({ showColorPicker = false }) => (
  <Uploader
    originalFile={new Blob()}
    fileName="my-image.png"
    fileSize={12345}
    downloadDisabled
    duotones={[
      {
        primary: '#D42D48',
        secondary: '#243261',
      }, {
        primary: '#D3F178',
        secondary: '#263661',
      }, {
        primary: '#FEFFFF',
        secondary: '#D5E5E8',
      },
    ]}
    primary="#D42D48"
    secondary="#243261"
    showColorPicker={showColorPicker}
    onDuotoneClick={action('onDuotoneClick')}
    onAddDuotoneClick={action('onAddDuotoneClick')}
    onSetPrimaryClick={action('onSetPrimaryClick')}
    onSetSecondaryClick={action('onSetSecondaryClick')}
    onToggleColorPicker={action('onToggleColorPicker')}
  />
);

storiesOf('Uploader', module)
  .add('simple example with Panel', () => (
    <Panel>
      <UploaderWithProps />
    </Panel>
  ))
  .add('simple example with Panel with new ColorPicker', () => (
    <Panel>
      <UploaderWithProps showColorPicker />
    </Panel>
  ))
  .add('simple example without Panel', () => (
    <UploaderWithProps />
  ));
