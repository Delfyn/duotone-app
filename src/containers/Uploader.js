/* @flow */

import { connect } from 'react-redux';
import {
  addDuotoneColor,
  setDuotoneColor,
  setPrimaryColor,
  setSecondaryColor,
  toggleColorPicker,
} from '../actions';
import Uploader from '../components/Uploader';

const mapStateToProps = state => ({
  originalFile: state.files.originalFile,
  fileName: state.files.fileName,
  fileSize: state.files.fileSize,
  downloadDisabled: !state.files.fileSize,
  duotones: state.colors.duotones,
  primary: state.colors.primary,
  secondary: state.colors.secondary,
  showColorPicker: state.ui.showColorPicker,
});

const mapDispatchToProps = dispatch => ({
  onDuotoneClick: (primary, secondary) => {
    dispatch(setDuotoneColor(primary, secondary));
  },
  onAddDuotoneClick: (primary, secondary) => {
    dispatch(addDuotoneColor(primary, secondary));
  },
  onSetPrimaryClick: (primary) => {
    dispatch(setPrimaryColor(primary));
  },
  onSetSecondaryClick: (secondary) => {
    dispatch(setSecondaryColor(secondary));
  },
  onToggleColorPicker: () => {
    dispatch(toggleColorPicker());
  },
});

const UploaderContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Uploader);

export default UploaderContainer;
