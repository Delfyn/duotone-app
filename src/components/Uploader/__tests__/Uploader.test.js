import React from 'react';
import renderer from 'react-test-renderer';
import Uploader from '../index';

test('Uploader renders correctly', () => {
  const component = renderer.create(<Uploader />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
