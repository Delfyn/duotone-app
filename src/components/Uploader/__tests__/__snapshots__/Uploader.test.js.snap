// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader renders correctly 1`] = `
<div>
  <header
    className="panel-file-picker flex items-center px2 py1"
    onClick={undefined}
  >
    <svg
      viewBox="24 0 72 72"
    >
      <path
        d="M60.493 72C79.883 72 96 55.882 96 36.493 96 16.118 79.882 0 60.493 0 40.118 0 24 16.118 24 36.493 24 55.883 40.118 72 60.493 72z"
        fill="#333"
        fillRule="evenodd"
      />
      <path
        d="M58 34h-9c-.553 0-1 .452-1 1.01v1.98c0 .567.448 1.01 1 1.01h9v9c0 .553.452 1 1.01 1h1.98c.567 0 1.01-.448 1.01-1v-9h9c.553 0 1-.452 1-1.01v-1.98c0-.567-.448-1.01-1-1.01h-9v-9c0-.553-.452-1-1.01-1h-1.98c-.567 0-1.01.448-1.01 1v9z"
        fill="#FFF"
        fillRule="evenodd"
      />
    </svg>
    <div
      className="mx2"
    >
      <h2
        className="h3 regular m0"
      >
        Choose image
      </h2>
    </div>
  </header>
  <main
    className="my2 px2 py2"
  >
    <p>
      Select a color combination or add your own to change the look of your photo.
    </p>
    <div>
      <svg
        className="cursor-pointer mr1 mb1"
        height={40}
        onClick={undefined}
        version="1.1"
        viewBox="0 0 40 40"
        width={40}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <circle
            cx={20}
            cy={20}
            id="path-1"
            r={20}
          />
          <mask
            fill="white"
            height={40}
            id="mask-2"
            maskContentUnits="userSpaceOnUse"
            maskUnits="objectBoundingBox"
            width={40}
            x="0"
            y="0"
          >
            <use
              xlinkHref="#path-1"
            />
          </mask>
          <polygon
            id="path-3"
            points="21 19 21 11 19 11 19 19 11 19 11 21 19 21 19 29 21 29 21 21 29 21 29 19"
          />
          <mask
            fill="white"
            height={20}
            id="mask-4"
            maskContentUnits="userSpaceOnUse"
            maskUnits="objectBoundingBox"
            width={20}
            x="0"
            y="0"
          >
            <use
              xlinkHref="#path-3"
            />
          </mask>
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
          transform="rotate(0 20 20)"
        >
          <g
            id="Artboard-2"
            stroke="#979797"
          >
            <g
              id="Plus"
            >
              <g
                id="Oval-3"
              >
                <use
                  mask="url(#mask-2)"
                  strokeWidth="4"
                  xlinkHref="#path-1"
                />
                <use
                  fill="#D8D8D8"
                  fillRule="evenodd"
                  id="Combined-Shape"
                  mask="url(#mask-4)"
                  strokeWidth="2"
                  xlinkHref="#path-3"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
  </main>
  <footer
    className="flex px2 pb2"
  >
    <button
      className="flex-auto btn"
      disabled={false}
      onClick={[Function]}
    >
      Download
    </button>
  </footer>
</div>
`;
